test/stdlib/PrintFloat.swift.gyb

```diff
@@ -0,0 +1,828 @@
+  expectDescription("125000.0", asFloat32(125000.0))
+  expectDescription("12500.0",  asFloat32(12500.0))
+  expectDescription("1250.0",   asFloat32(1250.0))
+  expectDescription("125.0",    asFloat32(125.0))
+  expectDescription("12.5",     asFloat32(12.5))
+  expectDescription("1.25",     asFloat32(1.25))
+  expectDescription("0.125",    asFloat32(0.125))
+  expectDescription("0.0125",   asFloat32(0.0125))
+  expectDescription("0.00125",  asFloat32(0.00125))
+  expectDescription("0.000125", asFloat32(0.000125))
+  expectDescription("1.25e-05", asFloat32(0.0000125))
+  expectDescription("1.25e-06", asFloat32(0.00000125))
+  expectDescription("1.25e-07", asFloat32(0.000000125))
+  expectDescription("1.25e-08", asFloat32(0.0000000125))
+  expectDescription("1.25e-09", asFloat32(0.00000000125))
+  expectDescription("1.25e-10", asFloat32(0.000000000125))
+  expectDescription("1.25e-11", asFloat32(0.0000000000125))
+  expectDescription("1.25e-12", asFloat32(0.00000000000125))
+  expectDescription("1.25e-13", asFloat32(0.000000000000125))
+  expectDescription("1.25e-14", asFloat32(0.0000000000000125))
+  expectDescription("1.25e-15", asFloat32(0.00000000000000125))
+  expectDescription("1.25e-16", asFloat32(0.000000000000000125))
+  expectDescription("1.25e-17", asFloat32(0.0000000000000000125))
+}
+
+PrintTests.test("Printable_Double") {
+  func asFloat64(_ f: Float64) -> Float64 { return f }
+
+  // Sanity check
+  let f = 100.125 as Double
+  expectEqual("f = 100.125", "f = \(f)")
+
+  expectDescription("0.0", asFloat64(0.0))
+  expectDescription("-0.0", asFloat64(-0.0))
+  expectDescription("0.1", asFloat64(0.1))
+  expectDescription("-0.1", asFloat64(-0.1))
+  expectDescription("1.0", asFloat64(1.0))
+  expectDescription("-1.0", asFloat64(-1.0))
+  expectDescription("1.1", asFloat64(1.1))
+  expectDescription("100.125", asFloat64(100.125))
+  expectDescription("-100.125", asFloat64(-100.125))
+
+  // Special values
+  expectDescription("3.141592653589793", Double.pi)
+  expectDescription("1.7976931348623157e+308", Double.greatestFiniteMagnitude)
+  expectDescription("5e-324", Double.leastNonzeroMagnitude)
+  expectDescription("2.2250738585072014e-308", Double.leastNormalMagnitude)
+  expectDescription("inf", Double.infinity)
+  expectDescription("-inf", -Double.infinity)
+  // Worst case for Double shortness:
+  expectDescription("2.311989689387339e-82", 2.311989689387339e-82)
+
+  // Verify NaNs
+  expectNaN("nan", Double.nan)
+  expectNaN("nan(0xffff)", Double(nan: 65535, signaling: false))
+  expectNaN("nan(0x3ffffffffffff)", Float64(bitPattern: 0x7fff_ffff_ffff_ffff))
+  expectNaN("nan(0x3ffffffffffff)", Float64(bitPattern: 0x7ffb_ffff_ffff_ffff))
+  expectNaN("-nan", -Double.nan)
+  expectNaN("-nan(0xffff)", -Double(nan: 65535, signaling: false))
+#if !arch(i386)
+  // 32-bit i386 lacks signaling Double nans
+  expectNaN("snan", Double.signalingNaN)
+  expectNaN("-snan", -Double.signalingNaN)
```

Comment:
@rintaro :  I'd like to add a test for `Double(nan: 0x4000000000001, signaling: true)`, but I'm not sure how that should print.  I think it should print as `"snan(0x4000000000001)"` but I think the previous code would print it as `"snan"` without the payload.

Reply:
As far as I can tell, the standard does not reserve any bit for signaling NaN.  It only requires that the quiet bit is not set and the payload is not zero.

I would expect to reserve a single value as the default signaling payload (not a particular bit), so that `Double(nan: 0x4_0000_0000_0000, signaling: true).debugDescription` produces `"snan"` but `Double(nan: 0x4_0000_0000_0001, signaling: true).debugDescription` produces `"snan(0x4000000000001)"`.

Reply:
> As far as I can tell, the standard does not reserve any bit for signaling NaN. It only requires that the quiet bit is not set and the payload is not zero.

It's weaker than that. There is no "quiet bit" in the standard. Some HW platforms use a "signaling bit" instead.

However, for platforms with the common-case behavior that the high-order significand bit is a quiet bit, Swift decided to canonicalize on `b10xxx...xxx` for qNaN and `b01xxx...xxx` for sNaN because this makes the same set of payloads encodable with each; otherwise quieting a sNaN is a lossy operation, which is undesirable.

(Note that we should actually switch to using `b11xxx...xxx` for qNaNs, so that `b00xxx...xxx` can serve as reserved bit patterns that are never generated by pure Swift operations on HW with the customary quiet bit).